lib/kubot/server.rb

```diff
@@ -0,0 +1,115 @@
+require 'http'
+require 'json'
+
+module Kubot
+  class MyServer < SlackRubyBot::Server
+
+    on :channel_joined do |client, data|
+      puts "client:",client, "data:", data
+      client.say(text: "Hello, I'm kubot I can help you with any issue.\nJust DM me or add me to the channel and tag me when you want some support!", channel: data.channel.id)
+      #explain how to communicate with Kubot
+    end 
+
+    on :message do |client, data|
+      if data.team == SLACK_SUPPORT_TEAM && data.bot_id.nil? 
```

Comment:
I didn't get it from code, why data.bot_id can be nil and what's stored there?

Reply:
If it's empty then the message wasn't sent from bot, so the channel can be created, otherwise it just constantly sends and forwards messages and never stops